replicator/src/main/scala/com/evolutiongaming/kafka/journal/replicator/Batch.scala

```diff
@@ -98,6 +109,15 @@ private[journal] object Batch {
                     val delete = deleteOf(a.to, b.origin orElse a.origin, b.version orElse a.version)
                     delete :: tail
                   }
+
+                  val cleanTail: List[Batch] = tail.map {
+                    case appends: Appends => dropDeleted(appends, a)
+                    case delete: Delete   => delete.some
+                    case purge: Purge     => purge.some
+                  }.flattenOption
```

Comment:
I think one can use `mapFilter` instead of `map+flattenOption` here.

Reply:
indeed, thank you!